### PR TITLE
feat: Extend TableHeader API for bg colors + revert box shadow

### PR DESCRIPTION
**Pull request by okeeffed**
## Objective
1. Enable the `TableHeader` to accept an optional prop to override the background color.
2. Revert the box shadow on the `TableCard` back to previous post-feedback.

### Context
Slack convos based on tickets:
- https://cultureamp.slack.com/archives/CETLTFG9G/p1588587636273700
- https://cultureamp.slack.com/archives/CETLTFG9G/p1588295755236700?thread_ts=1588295755.236700
- https://cultureamp.slack.com/archives/CHZQVHRK5/p1588295527203600
- https://cultureamp.slack.com/archives/CHZQVHRK5/p1588119044429700?thread_ts=1588118180.426800
- https://cultureamp.slack.com/archives/CHZQVHRK5/p1588553720324800?thread_ts=1588553720.324800
- https://cultureamp.slack.com/archives/CHZQVHRK5/p1588761923140000?thread_ts=1588760979.139100

JIRA tickets:
- https://cultureamp.atlassian.net/browse/FOW-1140
- https://cultureamp.atlassian.net/browse/FOW-1112

Figma designs:
- https://www.figma.com/file/cA1dCqQewJO5hMaIdKs4FS/Manager-task-widget?node-id=1098%3A226

### Causes
- The box shadow from token `$kz-shadow-small-box-shadow` bleeds through from the `TableCard` to child components. See the Slack thread for details, but this is an interim fix.

### Changes
1. Reverted the shadow in styles.
2. Extended the `TableHeader` API to take an optional `backgroundColor` that defaults the the current implementation of `ash` but enables overrides to accepted colors (only `white` for now).
3. Added a basic story with the prop override based on the Kaizen demo.
4. Added testing based on convention of checking valid classnames for the API.

### Notes

It looks as though any child button will maintain its darker grey colour if the header cell is selectable. Changes won't cause any breaking issues for current implementations, but could be weird down the track aesthetically if others opt for a white BG header and have selectable headers.

Screen Shot 2020-05-08 at 7 55 59 am

Screen Shot 2020-05-08 at 7 55 23 am

### Verification
1. Check the box shadow equates to the older box-shadow token.
2. Check that the default `TableHeader` has an ash background.
3. Check that the `TableHeader` can take a strongly typed `backgroundColor` prop that allows for `ash` or `white` and that `white` overrides the header.

### Checklist
* [x] I have updated the test suite
* [x] I have tested the change on Storybook
* [ ] I have tested the change locally
* [ ] I have done the cross-browser testing [supported browsers](https://academy.cultureamp.com/hc/en-us/articles/204539569-Supported-browsers-for-Participants)
* [x] I am basing off the correct branch
* [x] I have updated any relevant documentation or left useful comments in the code.
* [ ] I have reviewed the [security considerations](https://cultureamp.atlassian.net/wiki/spaces/AU/pages/923893881/Key+Security+Concepts) and highlighted them as appropriate.

### CODE REVIEW

File: draft-packages/table/KaizenDraft/Table/styles.scss [hunk 1]
```diff
@@ -148,6 +157,7 @@ $row-height: 60px;
   padding: 0 $ca-grid * 0.8;
   display: flex;
   align-items: center;
+  box-shadow: none;
```
Comment (mbylstra) [inline draft-packages/table/KaizenDraft/Table/styles.scss:160]: does this need to be added here? Cells already didn't have a box shadow, so it doesn't seem necessary to add this. Or is this to override where it's added somewhere else using a CSS selector? I'm lacking context on this change.

Comment (okeeffed) [reply draft-packages/table/KaizenDraft/Table/styles.scss:160]: oh geez not at all

Comment (okeeffed) [reply draft-packages/table/KaizenDraft/Table/styles.scss]: Sorry! Was a relic of playing around to see if it was the cell wreaking havoc!

**Comment by mbylstra** [comment]
Go for it, when you're ready to merge.

**Pull request merged**